plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package static
+
+import (
+	"encoding/json"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	ss "github.com/jaegertracing/jaeger/pkg/sampling/strategystore"
+	"github.com/jaegertracing/jaeger/thrift-gen/sampling"
+)
+
+type strategyStore struct {
+	logger *zap.Logger
+
+	defaultStrategy   *sampling.SamplingStrategyResponse
+	serviceStrategies map[string]*sampling.SamplingStrategyResponse
+}
+
+// NewStrategyStore creates a strategy store that holds static sampling strategies.
+func NewStrategyStore(options Options, logger *zap.Logger) (ss.StrategyStore, error) {
+	h := &strategyStore{
+		logger:            logger,
+		serviceStrategies: make(map[string]*sampling.SamplingStrategyResponse),
+	}
+	strategies, err := loadStrategies(options.StrategiesFile)
+	if err != nil {
+		return nil, err
+	}
+	h.parseStrategies(strategies)
+	return h, nil
+}
+
+// GetSamplingStrategy implements StrategyStore#GetSamplingStrategy.
+func (h *strategyStore) GetSamplingStrategy(serviceName string) (*sampling.SamplingStrategyResponse, error) {
+	if strategy, ok := h.serviceStrategies[serviceName]; ok {
+		return strategy, nil
+	}
+	return h.defaultStrategy, nil
+}
+
+// TODO good candidate for a global util function
```

Comment:
how much overlap is in this code with what the client is doing when parsing JSON response from the agent? Maybe we can combine that all into jaeger-lib?

Reply:
This TODO was originally meant for creating a util function for reading a json file and unmarshalling it. I didn't have as lofty goal of moving the client and this logic into jaeger-lib. I don't think it's similar enough to warrant that.